File: responses/AUS97P-NM9/longest_common_prefix.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def prefix_len(s, t)
  len = 0

  while len < s.size && len < t.size && s[len] == t[len]
    len += 1
  end

  len
end

def solution(a)
  a = a.sort
  prefix_len = (1...(a.size)).map { |i| prefix_len(a[i-1], a[i]) }.min

  a[0][0, prefix_len]
end

File: responses/SDXU35-KKU/longest_common_prefix.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

const sortByLength = (a, b) => a.length - b.length;

function solution(A) {
    let prefix = '';
    let sortedWords = A.sort(sortByLength);

    for (let i = sortedWords[0].length; i > 0; i--) {
        const testVal = sortedWords[0].substring(0, i);
        let test = true;
        for (word of A) {
            if (word.indexOf(testVal) !== 0) {
                test = false;
            }
        }
        if (test) {
            prefix = testVal;
            break;
        }
    }

    return prefix;
}
